### ISSUE TITLE
SMW should use Vue.js and Codex

### ISSUE THREAD

**Issue by krabina** [SemanticMediaWiki/SemanticMediaWiki#5546]
VueJS is the Wikimedia Foundation's official choice for adoption as the JavaScript framework for use within MediaWiki, available since MW 1.35.
Codex is the new design system for MediaWiki and will replace OOUI. Therfore we should talk about using it in SMW and related extensions.

- https://www.mediawiki.org/wiki/Vue.js
- https://www.mediawiki.org/wiki/Codex

Comments are welcome

**Comment by Nikerabbit** [comment]
At least for Codex to be usable, it would require waiting for a long time or reducing the number of supported MediaWiki versions. Or bundling Codex itself.

Vue might be more practical, but even though it's supported since 1.35, it's not the same version across all those releases.

On the other hand, I don't feel like the UI of SMW has big problems usability- or tech-wise. I'd prefer fixing scalability issues and addressing deprecation warnings.

**Comment by krabina** [comment]
I agree that there are more important issues, just wanted to open a thread about this as well since it might affect SMW at some point.

**Comment by alistair3149** [comment]
We have to figure out how to handle the compatibility issues with different MediaWiki version.

Like OOUI, each version of MediaWiki comes with a different version of Codex.
We probably can't use Codex fully in PHP at least until we move minimum MediaWiki version to 1.43 since CSS-only components and PHP support are not in 1.39. The most we can do is like #5786, we start converting existing UI to use Codex design tokens so that the UI ais at least more visually consistent with Codex.

**Comment by krabina** [comment]
Thank you for looking into this!
